src/Service/ProducerSideServiceImpl.cpp

```diff
@@ -53,6 +54,7 @@ void ProducerSideServiceImpl::OnCaptureStopRequested() {
     }
     LOG("About to send CaptureFinishedCommand to CaptureEventProducers (if any)");
     service_state_.capture_status = CaptureStatus::kCaptureFinished;
+    service_state_.capture_options = std::nullopt;
```

Comment:
Is there a concrete reason why you clear the options here?

Reply:
Hmm, not really, it just felt like the right thing to do, as they are only kept while they could actually be used (i.e., while `service_state_.capture_status != CaptureStatus::kCaptureFinished`).

Reply:
Okay, fine for me. Thanks!